backend/src/main/java/gov/cdc/usds/simplereport/service/ApiUserService.java

```diff
@@ -710,52 +683,45 @@ public UserInfo getUserByLoginEmail(String loginEmail) {
         throw new RestrictedAccessUserException();
       }
 
-      return consolidateUser(
-          apiUser,
-          oktaUser.getOrganizationRoleClaims(),
-          oktaUser.getStatus(),
-          oktaUser.isSiteAdmin());
+      return consolidateUser(apiUser, oktaUser);
     } catch (IllegalGraphqlArgumentException | UnidentifiedUserException e) {
       throw new OktaAccountUserException();
     }
   }
 
-  private UserInfo consolidateUser(
-      ApiUser apiUser,
-      Optional<OrganizationRoleClaims> optClaims,
-      UserStatus userStatus,
-      boolean isSiteAdmin) {
-
-    OrganizationRoleClaims claims = optClaims.orElseThrow(UnidentifiedUserException::new);
-
-    // use the target user's org so response is built correctly even if site admin is the requester
-    Organization org = _orgService.getOrganization(claims.getOrganizationExternalId());
-
-    List<Facility> facilities = _orgService.getFacilities(org);
-    Set<Facility> facilitiesSet = new HashSet<>(facilities);
-
-    boolean allFacilityAccess = claims.grantsAllFacilityAccess();
-    Set<Facility> accessibleFacilities =
-        allFacilityAccess
-            ? facilitiesSet
-            : facilities.stream()
-                .filter(f -> claims.getFacilities().contains(f.getInternalId()))
-                .collect(Collectors.toSet());
-
-    OrganizationRoles orgRoles =
-        new OrganizationRoles(org, accessibleFacilities, claims.getGrantedRoles());
-    if (!_featureFlagsConfig.isOktaMigrationEnabled() && !isSiteAdmin) {
-      try {
-        setRolesAndFacilities(orgRoles, apiUser);
-      } catch (PrivilegeUpdateFacilityAccessException e) {
-        log.warn(
-            "Could not migrate roles and facilities for user with id={} because facilities were invalid",
-            apiUser.getInternalId());
+  private UserInfo consolidateUser(ApiUser apiUser, PartialOktaUser oktaUser) {
+    boolean isSiteAdmin = oktaUser.isSiteAdmin();
+    UserStatus userStatus = oktaUser.getStatus();
+    OrganizationRoleClaims oktaClaims =
+        oktaUser.getOrganizationRoleClaims().orElseThrow(UnidentifiedUserException::new);
+    OrganizationRoles orgRoles = _orgService.getOrganizationRoles(oktaClaims);
+
+    if (!isSiteAdmin) {
+      if (_featureFlagsConfig.isOktaMigrationEnabled()) {
+        orgRoles = getOrgRolesFromDB(apiUser);
+      } else {
+        setRolesAndFacilities(Optional.ofNullable(orgRoles), apiUser);
       }
     }
+
     return new UserInfo(apiUser, Optional.of(orgRoles), isSiteAdmin, userStatus);
   }
 
+  private UserInfo consolidateNonSiteAdminUser(
```

Comment:
Reading this suite of methods, I think we could get away with just overloading / moving around some of the existing logic for `consolidateUser` rather than making a new method. That way, we just have the one interface into "stitch together API and Okta users" for both user types rather than duplicating the logic across the two. Let me know what you think of the following:

```
 private OrganizationRoles getOrgRoles(Optional<OrganizationRoleClaims> oktaClaims, ApiUser apiUser) {
        OrganizationRoles orgRoles = null;
        if (oktaClaims.isPresent()) {
            orgRoles = _orgService.getOrganizationRoles(oktaClaims.get());
        }
        if (_featureFlagsConfig.isOktaMigrationEnabled()) {
            orgRoles = getOrgRolesFromDB(apiUser);
        } else {
            setRolesAndFacilities(Optional.ofNullable(orgRoles), apiUser);
        }
        return orgRoles;
    }

    private UserInfo consolidateUser(ApiUser apiUser, PartialOktaUser oktaUser) {
        OrganizationRoleClaims orgRoleClaims = oktaUser.getOrganizationRoleClaims().orElseThrow(UnidentifiedUserException::new);
        UserStatus userStatus = oktaUser.getStatus();
        boolean isSiteAdmin = oktaUser.isSiteAdmin();
        OrganizationRoles orgRoles = getOrgRoles(Optional.of(orgRoleClaims), apiUser);
        return new UserInfo(apiUser, Optional.of(orgRoles), isSiteAdmin, userStatus);
    }

    private UserInfo consolidateUser(
            ApiUser apiUser, Optional<OrganizationRoleClaims> oktaClaims) {
        OrganizationRoles orgRoles = getOrgRoles(oktaClaims, apiUser);
        return new UserInfo(apiUser, Optional.of(orgRoles), false);
    }
```

Reply:
I didn't overload the `consolidateUser` method because I wanted to make obvious within the method name that `consolidateNonSiteAdminUser` shouldn't be called for a site admin in case this method is used in the future to get a user's `UserInfo` and before the migration changes are cleaned up.

I could also leave this warning in a doc comment for that version of the method specifically, if the preference is to consolidate these methods. Let me know what you think!

Reply:
I see your concern here, but as is, are we splitting out the conditions between site admins and non-site admins? For instance, you could call `consolidateUser` on a non-site admin. That means for those non site admins, both `consolidateUser` and `consolidateNonSiteAdmin` are valid, which feels confusing. 

I think the actual difference between these methods are whether we're just passing in the claims or the the claims plus the extra info within `PartialOktaUser`. That leads me to think that overloading this method might be a good call since the underlying action (consolidating the user) is pretty similar in both cases. It would also bypass the issue for the caller needing to know about the admin status of the user, which feels like a good abstraction since we're checking for site admin status in the method internals anyways.

Reply:
Updated this with a slight modification to include a `isSiteAdmin` param so we don't try to save or fetch roles in the DB for siteAdmins. Let me know what you think, Bob!